gateway-provider-security-pac4j/src/test/java/org/apache/knox/gateway/pac4j/filter/Pac4jDispatcherFilterTest.java

```diff
@@ -0,0 +1,161 @@
+     */
+    @Test
+    public void testDefaultCookieMaxAge() throws Exception {
+        String COOKIE_MAX_AGE = "-1";
+        List<String> params = new ArrayList<>();
+        params.add(Pac4jDispatcherFilter.PAC4J_CALLBACK_URL);
+        params.add("clientName");
+        params.add(SAML_KEYSTORE_PATH);
+        params.add(SAML_IDENTITY_PROVIDER_METADATA_PATH);
+
+        KeyStore ks = KeyStore.getInstance("JKS");
+
+        ServletContext context = EasyMock.createNiceMock(ServletContext.class);
+        GatewayServices services = EasyMock.createNiceMock(GatewayServices.class);
+        CryptoService cryptoService = EasyMock.createNiceMock(CryptoService.class);
+        AliasService aliasService = EasyMock.createNiceMock(AliasService.class);
+        KeystoreService keystoreService = EasyMock.createNiceMock(KeystoreService.class);
+        MasterService masterService = EasyMock.createNiceMock(MasterService.class);
+        FilterConfig filterConfig = EasyMock.createNiceMock(FilterConfig.class);
+
+        EasyMock.expect(keystoreService.getKeystoreForGateway()).andReturn(ks).anyTimes();
+        EasyMock.expect(masterService.getMasterSecret()).andReturn("apacheknox".toCharArray()).anyTimes();
+
+        EasyMock.expect(context.getAttribute(GatewayServices.GATEWAY_SERVICES_ATTRIBUTE)).andReturn(services).anyTimes();
+        EasyMock.expect(context.getAttribute(GatewayServices.GATEWAY_CLUSTER_ATTRIBUTE)).andReturn(TEST_CLUSTER_NAME).anyTimes();
+        EasyMock.expect(services.getService(ServiceType.CRYPTO_SERVICE)).andReturn(cryptoService).anyTimes();
+        EasyMock.expect(services.getService(ServiceType.ALIAS_SERVICE)).andReturn(aliasService).anyTimes();
+        EasyMock.expect(services.getService(ServiceType.KEYSTORE_SERVICE)).andReturn(keystoreService).anyTimes();
+        EasyMock.expect(services.getService(ServiceType.MASTER_SERVICE)).andReturn(masterService).anyTimes();
+        EasyMock.expect(filterConfig.getServletContext()).andReturn(context).anyTimes();
+        EasyMock.expect(filterConfig.getInitParameterNames()).andReturn(Collections.enumeration(params)).anyTimes();
+        EasyMock.expect(filterConfig.getInitParameter(Pac4jDispatcherFilter.PAC4J_CALLBACK_URL)).andReturn(TEST_CALLBACK_URL).anyTimes();
+        EasyMock.expect(filterConfig.getInitParameter(SAML_KEYSTORE_PATH)).andReturn("/var/keystore").anyTimes();
+        EasyMock.expect(filterConfig.getInitParameter(SAML_IDENTITY_PROVIDER_METADATA_PATH)).andReturn("/tmp/sp-metadata.xml").anyTimes();
+        EasyMock.expect(filterConfig.getInitParameter("clientName")).andReturn("SAML2Client").anyTimes();
+        EasyMock.expect(aliasService.getPasswordFromAliasForCluster(TEST_CLUSTER_NAME, KnoxSessionStore.PAC4J_PASSWORD, true))
+                .andReturn(KnoxSessionStore.PAC4J_PASSWORD.toCharArray()).anyTimes();
+
+
+        EasyMock.replay(context, services, cryptoService, aliasService, keystoreService, masterService, filterConfig);
```

Comment:
These are common expectations for both of the test cases, you may want them in a private method to avoid code duplication.